src/ServerDaemon.chpl

```diff
@@ -724,6 +736,30 @@ module ServerDaemon {
                                                                             elapsed));
             this.shutdown(); 
         }
+
+        /* While waiting for a message from client, do checkpointing if desired. */
+        proc startIdleServer() throws {
+          // Do nothing if the checkpointing module was not included.
+          if commandMap.contains("auto_checkpoint") {
+            const msgArgs = new owned MessageArgs();
+            msgArgs.addPayload(b"idle start");
```

Comment:
Is the motivation for using a message-based approach to be able to do `commandMap.contains`? I am on board with it in any case. However, checkpointing is a core feature of Arkouda and we can consider making it part of the server no matter what. I can't imagine it adding a ton of compilation time.

Short of that, it would be good to wire in a mechanism so that the server can check whether a module is loaded or not without having to do this. I also wanted that feature so that when the server sees a message to be not supported, it can tell you which module it belongs to. IOW, during compilation, we would still take a look at the excluded modules just to see which messages they implement, but then not compile them. That's a bit of a tangent, though.